contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+     * @param user address of the user
+     * @param price price
+     */
+    function _validateERC20(
+        address currency,
+        address user,
+        uint256 price
+    ) internal view returns (uint256 validationCode) {
+        if ((IERC20(currency).allowance(user, address(looksRareExchange))) < price)
+            return ERC20_APPROVAL_INFERIOR_TO_AMOUNT;
+        if ((IERC20(currency).balanceOf(user)) < price) return ERC20_BALANCE_INFERIOR_TO_AMOUNT;
+    }
+
+    /**
+     * @notice Check validity of ERC721 approvals and balances required to process the order
+     * @param collection address of the collection
+     * @param user address of the user
+     * @param transferManager address of the transfer manager
+     * @param tokenId tokenId
+     */
+    function _validateERC721AndEquivalents(
+        address collection,
+        address user,
+        address transferManager,
+        uint256 tokenId
+    ) internal view returns (uint256 validationCode) {
+        (bool answer, bytes memory data) = collection.staticcall(
+            abi.encodeWithSelector(IERC721.getApproved.selector, tokenId)
+        );
+
+        address approvedAddress;
+
+        if (answer) {
+            approvedAddress = abi.decode(data, (address));
+        }
+
+        bool isApprovedSingle = approvedAddress == transferManager;
+
+        (answer, data) = collection.staticcall(
+            abi.encodeWithSelector(IERC721.isApprovedForAll.selector, user, transferManager)
+        );
+
+        bool isApprovedAll;
+
+        if (answer) {
+            isApprovedAll = abi.decode(data, (bool));
+        }
+
+        if (!isApprovedAll && !isApprovedSingle) return ERC721_NO_APPROVAL_FOR_ALL_OR_TOKEN_ID;
+        if ((IERC721(collection).ownerOf(tokenId)) != user) return ERC721_TOKEN_ID_NOT_IN_BALANCE;
+    }
+
+    /**
+     * @notice Check validity of ERC1155 approvals and balances required to process the order
+     * @param collection address of the collection
+     * @param user address of the user
+     * @param transferManager address of the transfer manager
+     * @param tokenId tokenId
+     * @param amount amount
+     */
+    function _validateERC1155(
+        address collection,
+        address user,
+        address transferManager,
+        uint256 tokenId,
+        uint256 amount
+    ) internal view returns (uint256 validationCode) {
+        bool isApprovedAll = IERC1155(collection).isApprovedForAll(user, transferManager);
+        if (!isApprovedAll) return ERC1155_NO_APPROVAL_FOR_ALL;
+        if ((IERC1155(collection).balanceOf(user, tokenId)) < amount)
+            return ERC1155_BALANCE_TOKEN_ID_INFERIOR_TO_AMOUNT;
+    }
+
+    /**
+     * @notice Check validity of EOA maker order
+     * @param digest digest
+     * @param targetSigner the signer address to confirm message validity
+     * @param v parameter (27 or 28). This prevents maleability since the public key recovery equation has two possible solutions.
```

Comment:
```suggestion
     * @param v parameter (27 or 28). This prevents malleability since the public key recovery equation has two possible solutions.
```